internal/driver/reservedip.go

```diff
@@ -1,126 +1,44 @@
 package driver
 
 import (
-	"context"
-	"encoding/json"
-	"fmt"
+	log "github.com/sirupsen/logrus"
 
 	"github.com/coreos/etcd/clientv3"
-	"github.com/coreos/etcd/mvcc/mvccpb"
-	"github.com/pkg/errors"
+	"github.com/projecteru2/minions/lib"
 )
 
 const fixedIPLabel = "fixed-ip"
 const inheritLabel = "ip-inherit"
 
-type reservedIPAddress struct {
-	Address   string
-	Allocated bool
-}
-
-// IPAddressManager .
-type ReservedIPManager interface {
-	MarkReservedIP(ctx context.Context, ipAddress string) error
-	CheckAndAquireReservedIP(ctx context.Context, ipAddress string) (bool, error)
-	IsReservedIP(ctx context.Context, ipAddress string) (bool, error)
-	ReleaseReservedIP(ctx context.Context, ipAddress string) error
-}
-
 type reservedIPManager struct {
-	etcdV3 *clientv3.Client
-}
-
-func pathToIPAddress(ipAddress string) string {
-	return fmt.Sprintf("/barrel/reserved/ip/%s", ipAddress)
+	etcd lib.EtcdClient
 }
 
-func kvToModel(ekv *mvccpb.KeyValue) (reservedIPAddress, error) {
-	result := reservedIPAddress{}
-	err := json.Unmarshal(ekv.Value, &result)
-	return result, err
-}
-
-func modelToKv(model reservedIPAddress) (string, string) {
-	key := pathToIPAddress(model.Address)
-	// adopt json.Marshal will introduce handling error, so use json template
-	return key, fmt.Sprintf(`{"Address":"%s", "Allocated":%v}`, model.Address, model.Allocated)
+// ReservedIPManager .
+type ReservedIPManager interface {
+	ReserveIP(ip string, containerID string) error
+	AquireIPIfReserved(ip string) (bool, error)
```

Comment:
like aforementioned, the ```IPIfReserved``` seems no need here as the ```ReservedIPManager``` name, I consider the ```ReservedIPManager``` shouldn't Reserve/Acquire any IP that doesn't mark as ```Reserved```, so the ```IfReserved``` like an implicit condition.

Reply:
IfReserved used here is indicating that calling this method will not encounter an error when the ip is not reserved.